third_party/move/move-compiler-v2/src/pipeline/explicit_drop.rs

```diff
@@ -0,0 +1,197 @@
+use super::{
+    livevar_analysis_processor::{LiveVarAnnotation, LiveVarInfoAtCodeOffset},
+    reference_safety_processor::{LifetimeAnnotation, LifetimeInfoAtCodeOffset},
+};
+use move_binary_format::file_format::CodeOffset;
+use move_model::{ast::TempIndex, model::FunctionEnv, ty::Type};
+use move_stackless_bytecode::{
+    function_target::{FunctionData, FunctionTarget},
+    function_target_pipeline::{FunctionTargetProcessor, FunctionTargetsHolder},
+    stackless_bytecode::{AssignKind, AttrId, Bytecode, Operation},
+};
+use std::collections::BTreeSet;
+
+pub struct ExplicitDrop {}
+
+impl FunctionTargetProcessor for ExplicitDrop {
+    fn process(
+        &self,
+        _targets: &mut FunctionTargetsHolder,
+        fun_env: &FunctionEnv,
+        mut data: FunctionData,
+        _scc_opt: Option<&[FunctionEnv]>,
+    ) -> FunctionData {
+        if fun_env.is_native() {
+            return data;
+        }
+        let target = FunctionTarget::new(fun_env, &data);
+        let mut transformer = ExplicitDropTransformer::new(target);
+        transformer.transform();
+        data.code = transformer.transformed;
+        data.annotations.remove::<LiveVarAnnotation>();
+        data.annotations.remove::<LifetimeAnnotation>();
+        data
+    }
+
+    fn name(&self) -> String {
+        "ExplicitDrop".to_owned()
+    }
+}
+
+struct ExplicitDropTransformer<'a> {
+    target: FunctionTarget<'a>,
+    // result of the transformation
+    transformed: Vec<Bytecode>,
```

Comment:
Consider using the same transformation style as copy transformation in livevar_analysis_processor. You can also avoid cloning the bytecode in `transform` and instead use the consuming `into_iter` (also, we would be consistent in our transformations).

Reply:
Added to new PR.